lib/veri_policy.mli

```diff
@@ -0,0 +1,30 @@
+open Core_kernel.Std
+open Bap.Std
+open Bap_traces.Std
+
+(** rule = ACTION : INSN : EVENT : EVENT  *)
+
+type t
+type event = Trace.event
+type events = Value.Set.t
+type set_pair = events * events
+type action
+
+val skip : action
+val deny : action
+
+val create:
+  ?insn:string -> ?left:string -> ?right:string -> action -> t
+
+val match_events: t -> string -> events -> events -> 
+  (event option * event option) list
+
+(** type [deny_error] list all events that matches to deny rule *)
+type deny_error = set_pair
+
+type r = (set_pair, deny_error) Result.t
```

Comment:
As a second thought, `Either` also doesn't fit very well. This `process` function should be considered closer. We apply policy to two sets of events. As a result of policy application is the set of rules, that were denied. So the result type is just a set of events. And the better name is `denied`. 
The fact, that your event has "successfully filtered events set" means, that you misunderstood the algorithm. There should no such event. And "filtered" in that scope doesn't have any meaning.